File: src/pybend/static/NTT0.3/utils/Logging.js

export default class Logging {
    static log(val1, val2) {
        let caller = log.caller().name
        console.log(val1, val2, caller)
    }
   static dev(val1, val2) {
       console.log(val1, val2)
   }

    static debug(val1, val2) {
        console.info(val1, val2)
    }

   static warn(val1, val2) {
       if (val2)
           console.warn(val1, val2)
       else
           console.warn(val1)
   }

   static error(val1, val2) {
       if (!val2)
           console.error("Logging error:", val1)
       else
           console.error(val1, val2)
       //console.error(val2)
   }
}

File: packages/n3tx-core/src/n3tx_core/utils/logging.py
"""Logging configuration for N3TX applications.

Sets up the root logger to use Uvicorn's colourized, aligned formatter so that
application and framework log output is visually consistent with Uvicorn's own
access/error logs::

    INFO:     n3tx.storage: Database opened
    WARNING:  veille.scheduler: Run queue is full
    ERROR:    n3tx.agents: Tool discovery failed
"""

import logging
import sys


def setup_logging(level: int = logging.INFO) -> None:
    """Configure the root logger with Uvicorn's colourized formatter.

    Should be called once at application startup (``N3TXApp.__init__`` does
    this automatically).  Calling it again is safe — ``force=True`` replaces
    existing handlers.

    Args:
        level: Root log level.  Defaults to ``logging.INFO``.
    """
    try:
        from uvicorn.logging import DefaultFormatter
        fmt = DefaultFormatter('%(levelprefix)s %(name)s: %(message)s', use_colors=None)
    except ImportError:  # pragma: no cover — uvicorn always present at runtime
        fmt = logging.Formatter('%(levelname)-8s %(name)s: %(message)s')

    handler = logging.StreamHandler(sys.stderr)
    handler.setFormatter(fmt)
    logging.basicConfig(level=level, handlers=[handler], force=True)
